profiles/schema.py

```diff
@@ -874,16 +874,26 @@ class Arguments:
     @staticmethod
     def _handle_primary_email(profile, primary_email_input):
         if primary_email_input:
-            profile.emails.update(primary=False)
-
             email_address = primary_email_input["email"]
-            email = Email(
-                profile=profile,
-                email=email_address,
-                email_type=EmailType.NONE,
-                primary=True,
-            )
-            email.save()
+
+            email = profile.emails.filter(email=email_address).first()
+
+            if email:
+                if not email.primary:
+                    email.primary = True
+                else:
+                    email = None
+            else:
+                email = Email(
+                    profile=profile,
+                    email=email_address,
+                    email_type=EmailType.NONE,
+                    primary=True,
+                )
+
+            if email:
+                profile.emails.filter(primary=True).update(primary=False)
+                email.save()
```

Comment:
I would try to make this a bit clearer. Maybe like:

```python
email, email_created = profile.emails.get_or_create(email=email_address, defaults={
    "email_type": EmailType.NONE,
})

profile.emails.exclude(pk=email.pk).update(primary=False)

if not email.primary:
    email.primary = True
    email.save()
```

I didn't test this, but something like this should work.

Also I'm not a big fan of always calling this method and then checking if an email was provided. At the very least the check could be

```python
if not primary_email_input:
    return
```

Reply:
> Also I'm not a big fan of always calling this method and then checking if an email was provided.

So, would it be better if the checking would take place at the calling site, before this function is called?

Reply:
I think so yes. Then when reading that function you would know if something will happen to the emails or not. And don't need to jump to this function to check.

Reply:
Updated now as suggested, as it was a good one. The `mutate` function was getting too complex (according to flake8) so I inserted the commit a44724b to make it simpler.